From 1a404b68e8a6d34d8728d08962e9a84a23f58e23 Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Sun, 6 Sep 2015 18:38:01 +0900
Subject: [PATCH] uupdate: exit code is tricky

Bug:
 * If SVERSION=1-1, uupdate works. (right)
 * If SVERSION=1, uupdate thinks it is a native Debian package (right).
 * If SVERSION=0-1, uupdate thinks it is a native Debian package (wrong).

The reason is the return code of expr is:
       Exit status is 0 if EXPRESSION is neither null nor 0, 1  if  EXPRESSION
       is  null  or  0,  2 if EXPRESSION is syntactically invalid, and 3 if an
       error occurred.

If SVERSION=1, it is null and return 1.  Good.
If SVERSION=0-1, it is 0  and return 1.   Not nice!
---
 scripts/uupdate.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/uupdate.sh b/scripts/uupdate.sh
index e037327..4240ea8 100755
--- a/scripts/uupdate.sh
+++ b/scripts/uupdate.sh
@@ -237,11 +237,11 @@ mustsetvar VERSION "`dpkg-parsechangelog -SVersion`" "source version"
 # Get epoch and upstream version
 eval `echo "$VERSION" | perl -ne '/^(?:(\d+):)?(.*)/; print "SVERSION=$2\nEPOCH=$1\n";'`
 
-UVERSION=`expr "$SVERSION" : '\(.*\)-[0-9a-zA-Z.+~]*$'` ||
-{
+UVERSION=`expr "$SVERSION" : '\(.*\)-[0-9a-zA-Z.+~]*$' || true`
+if [ -z "$UVERSION" ]; then
     echo "$PROGNAME: a native Debian package cannot take upstream updates" >&2
     exit 1
-}
+fi
 
 # Save pwd before we goes walkabout
 OPWD=`pwd`
-- 
2.1.4

